### PR TITLE
Shortcut SF condition when canon known

### PR DESCRIPTION
**Pull request by gavofyork**

### CODE REVIEW

File: ethcore/src/client/mod.rs [hunk 1]
```diff
@@ -228,6 +228,11 @@ pub trait BlockChainClient : Sync + Send {
 
 	/// Get `Some` gas limit of block 1_760_000, or `None` if chain is not yet that long.
 	fn dao_rescue_block_gas_limit(&self, chain_hash: H256) -> Option<U256> {
+		// shortcut if the canon chain is already known.
+		if self.chain_info().best_block_number > 1_761_000 {
+			return self.block_header(BlockID::Number(1_760_000)).map(|header| HeaderView::new(&header).gas_limit());
+		}
+		// otherwise check according to `chain_hash`.
```
Comment (debris) [inline ethcore/src/client/mod.rs]: why not just?

```rust
fn dao_rescue_block_gas_limit(&self) -> Option<U256> {
  self.block_header(BlockID::Number(1_760_000)).map(|header| HeaderView::new(&header).gas_limit())
}
```

the code below is doing exactly the same thing. It's also less efficient and more complex

Comment (tomusdrw) [inline ethcore/src/client/mod.rs]: Your code will look only in canon chain. The code below is looking for the block in other possible chains.

Comment (debris) [inline ethcore/src/client/mod.rs]: so this make the pr code invalid, cause it cares about the canon chain

Comment (tomusdrw) [inline ethcore/src/client/mod.rs]: Why? The first part is shortcutting if canon is more then 1000 blocks after the trigger. The code below will be run between `1_760_000` and `1_761_000`.

**Comment by debris** [comment]
My bad. `1_761_000` looks almost the same as `1_760_000`. It would be good to make constants from these numbers.

File: ethcore/src/client/mod.rs (hunk 1 above)
Comment (debris) [inline ethcore/src/client/mod.rs]: My bad. `1_761_000` looks almost the same as `1_760_000`. It would be good to make constants from these numbers.

**Comment by gavofyork** [comment]
the code will be removed in 2 days; i think we all understand what it does now :)

**Comment by arkpar** [comment]
Build fails. Needs rebasing probably

**Comment by gavofyork** [comment]
builds & tests fine locally.

**Pull request merged**